Test failure: TestHttpMisc.test_html

I ran `tox` in a git checkout and saw one failure;

```
________________________________________ TestHttpMisc.test_html _________________________________________
tests/__init__.py:95: in newfunc
    return func(*args, **kwargs)
tests/checker/test_http_misc.py:32: in test_html
    self.obfuscate_test()
tests/checker/test_http_misc.py:68: in obfuscate_test
    self.direct(url, resultlines, recursionlevel=0)
tests/checker/__init__.py:266: in direct
    self.fail_unicode(unicode(os.linesep).join(l))
tests/checker/__init__.py:243: in fail_unicode
    self.fail(msg)
E   AssertionError: Differences found testing http://0xc1639055/
E   @@ -1,6 +1,6 @@
E    url http://0xc1639055/
E    cache key http://193.99.144.85/
E   -real url http://193.99.144.85/
E   -info Access denied by robots.txt, checked only syntax.
E   +real url https://www.heise.de/
E   +info Redirected to `https://www.heise.de/'.
E    warning URL http://0xc1639055/ has obfuscated IP address 193.99.144.85
E    valid
```

It looks like the test is making actual HTTP requests to an actual HTTP server not under our control, and expects that external HTTP server not under our control to behave in a certain way.

I do not like such tests.

Options:
1. change the test's expectations to match what heise.de does today (redirect to HTTPS, very sensible)
2. rewrite the test to talk to an HTTP server under our control (ideally running on localhost:randomport, then we can have the test try to talk to http://0x7f000001:port/ -- since AFAIU the purpose of this test is to see if linkchecker understands obfuscated ways of representing an IP address).
3. rewrite the test to be more of a unit test rather than a functional test, so we can see that whatever the bits of code that parse URLs recognize this one and emits the warning without actually seeing how linkchecker connects to the remote server

I do not have the motivation for options 2 or 3 at the moment.

I have also [configured Travis CI]()https://travis-ci.org/linkchecker/linkchecker/settings) to run daily tests of master so we can notice test breakages due to such external changes sooner, and not when someone creates a new PR.

--- a/tests/checker/test_http_misc.py
+++ b/tests/checker/test_http_misc.py
@@ -60,8 +60,8 @@ class TestHttpMisc (HttpServerTest):
         resultlines = [
             u"url %s" % url,
             u"cache key %s" % rurl,
-            u"real url %s" % rurl,
-            u"info Access denied by robots.txt, checked only syntax.",
+            u"real url https://%s/" % host,
+            u"info Redirected to `https://%s/'." % host,
             u"warning URL %s has obfuscated IP address %s" % (url, ip),
             u"valid",
         ]

